### PR TITLE
Better doors (gates)

### LINKED ISSUE #62: TODO: turn off DoorClose and DoorOpen animation "Loop time"
This is a bug I didn't notice originally because I didn't have players and enemies to test with.

We need to turn off the looping or else the doors will not remain open once they open, for example.

### PR DESCRIPTION
**Pull request by AleksandrHovhannisyan**
- Fixed #62 bug
- Created gate prefab

### CODE REVIEW

**Comment by klein1992** [comment]
I killed the enemy in your sandbox put no gate opening occurred :( Also, please remove the extra shoot controller. The player had 2 shoot controllers for some reason

File: Assets/Scripts/Entity/Enemy/Enemy.cs [hunk 1]
```diff
@@ -51,6 +51,8 @@ public override void ChangeMaxHealthBy(int amount)
      */
     public override void ChangeHealthBy(int amount)
     {
+        if (isDead) return;
```
Comment (klein1992) [inline Assets/Scripts/Entity/Enemy/Enemy.cs:54]: Fail Fast....I like dis. Jazy approves

File: Assets/Scripts/Environmental/Room/RoomScript.cs [hunk 2]
```diff
@@ -23,11 +23,9 @@ private void Awake()
         numEnemies = spawnedEnemies.Count;
         SubscribeToEnemyDeath();
 
-        theme = gameObject.GetComponent<AudioSource>();
-
-        if(theme.clip != null)
+        if(audioTheme != null && audioTheme.clip != null)
```
Comment (klein1992) [inline Assets/Scripts/Environmental/Room/RoomScript.cs:26]: could use Failfast here...

**Comment by AleksandrHovhannisyan** [comment]
I didn't add any extra shoot controller. No gate opening occurred because you probably didn't assign the door to the room.

**Comment by klein1992** [comment]
Ah....don't know who did but there were 2 shoot controllers on main player. Will assign doors and review in a sec.

**Comment by klein1992** [comment]
Peachy AF dude.

**Review by klein1992 — APPROVED** [review]

**Pull request merged**